pallets/portfolio/src/lib.rs

```diff
@@ -221,49 +232,33 @@ decl_module! {
         /// # Errors
         /// * `PortfolioDoesNotExist` if `num` doesn't reference a valid portfolio.
         /// * `PortfolioNotEmpty` if the portfolio still holds any asset
-        #[weight = 1_000_000_000]
+        #[weight = <T as Trait>::WeightInfo::delete_portfolio()]
         pub fn delete_portfolio(origin, num: PortfolioNumber) -> DispatchResult {
             let PermissionedCallOriginData {
                 primary_did,
                 secondary_key,
                 ..
             } = Identity::<T>::ensure_origin_call_permissions(origin)?;
 
-            if let Some(sk) = secondary_key {
-                // Check that the secondary signer is allowed to work with this portfolio.
-                ensure!(
-                    sk.has_portfolio_permission(iter::once(num)),
-                    Error::<T>::SecondaryKeyNotAuthorizedForPortfolio
-                );
-            }
-            // Check that the portfolio exists.
-            ensure!(
-                <Portfolios>::contains_key(&primary_did, &num),
-                Error::<T>::PortfolioDoesNotExist
-            );
```

Comment:
This logic is duplicated in the pallet elsewhere; could you refactor into an `ensure_...` function?